### PR TITLE
[PVR] PVRMananger::m_currentFile: CFileItem* -> CFileItemPtr.

### PR DESCRIPTION
**Pull request by ksooo**
Cleanup: Replace raw pointer with smart pointer.

@Jalle19 for review?

### CODE REVIEW

File: xbmc/pvr/PVRManager.cpp [hunk 1]
```diff
@@ -1579,9 +1571,8 @@ bool CPVRManager::PerformChannelSwitch(const CPVRChannelPtr &channel, bool bPrev
 
   CLog::Log(LOGDEBUG, "PVRManager - %s - switching to channel '%s'", __FUNCTION__, channel->ChannelName().c_str());
 
-  // will be deleted by CPVRChannelSwitchJob::DoWork()
```
Comment (Jalle19) [inline xbmc/pvr/PVRManager.cpp]: Did you change this usage too? Can't really see from the diff.

**Review by Jalle19 — APPROVED** [review]
Apart from the comment, +1

File: xbmc/pvr/PVRManager.cpp (hunk 1 above)
Comment (ksooo) [inline xbmc/pvr/PVRManager.cpp]: Hehe, good catch, <code>CPVRChannelSwitchJob::DoWork</code> actually did not (!) delete it, this was a bug, which now is fixed.

Comment (ksooo) [inline xbmc/pvr/PVRManager.cpp]: CPVRChannelSwitchJob now also uses shared pointers.

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
WIN-32 Jenkins error is not related to this PR.

**Pull request merged**